### ISSUE TITLE
React table filter overlaps on First row.

### ISSUE THREAD

**Issue by sagarjpawar** [GuillaumeJasmin/react-table-hoc-fixed-columns#33]
![image](https://user-images.githubusercontent.com/3666279/57537560-2c612a80-7364-11e9-84e0-4a4b0bfc4154.png)
Hi,
Thanks for the fixed table HOC.It is really useful.
However,I am facing issue when hiding column dynamically.
https://codesandbox.io/s/9y7oolr85p.

I have kept a button "Show/Hide column" at the bottom of the table.After clicking on the button,it hide the "Age test" column which overlaps the filter on first row.

After inspecting the element, filter div contain style top: 45px.

If i directly use the react-table, it's working fine.However while using Fixed HOC, facing this issue.
https://codesandbox.io/s/wow8omql1l

Can you please suggest me available HOC which will fix this issue.

**Comment by GuillaumeJasmin** [comment]
Hi,
Thanks for your feedback.
You should upgrade to v2, that will fix your bug

**Comment by sagarjpawar** [comment]
Hi,
After upgrading the version, fixed columns are not working.please refer to the same sandbox URL.

**Comment by GuillaumeJasmin** [comment]
Please, read release note...

**Comment by sagarjpawar** [comment]
Hi,
Thank you for quick response.
As per release note,I have imported the css('react-table-hoc-fixed-columns/lib/styles.css').However,I can see the previous issue again.There is a space between the header and filter search box after hiding column.

Please let me know if i miss anything.
Please refer the same sandbox url.

**Comment by tscott2** [comment]
I'm having the same issue. Did you find a resolution?

**Comment by tscott2** [comment]
I see the problem. In src/lib/stickyPosition/index.js the css top position is being set, but it doesn't get set correctly, I think especially when the height changes. So I think the code: 

```
    componentDidMount() {
      const headerRows = document.querySelectorAll(`.${this.uniqClassName} .rt-thead`);
      /* eslint-disable no-param-reassign */
      Array.from(headerRows).forEach((row) => {
        row.style.top = `${row.offsetTop}px`;
      });
      /* eslint-enable no-param-reassign */
    }
```

should be replaced with:


```
    componentDidMount() {
      const headerRows = document.querySelectorAll(`.${this.uniqClassName} .rt-thead`);
      let topPosition = 0;
      /* eslint-disable no-param-reassign */
      Array.from(headerRows).forEach((row) => {
        row.style.top = `${topPosition}px`;
        topPosition += row.offsetHeight;
      });
      /* eslint-enable no-param-reassign */
    }

    componentDidUpdate() {
      const headerRows = document.querySelectorAll(`.${this.uniqClassName} .rt-thead`);
      let topPosition = 0;
      /* eslint-disable no-param-reassign */
      Array.from(headerRows).forEach((row) => {
        row.style.top = `${topPosition}px`;
        topPosition += row.offsetHeight;
      });
      /* eslint-enable no-param-reassign */
    }
```

or something similar.

**Comment by GuillaumeJasmin** [comment]
Hi @tscott2 ,
Thanks for your feedback and providing the suggestion.
Your code works well.

It's fixed on `v2.1.1`
